etherman/etherman_signers.go

```diff
@@ -0,0 +1,69 @@
+package etherman
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/agglayer/go_signer/signer"
+	signertypes "github.com/agglayer/go_signer/signer/types"
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/ethereum/go-ethereum/core/types"
+)
+
+// EthermanSigners is a struct that holds the signers
+type EthermanSigners struct {
+	chainID uint64
+	signers map[common.Address]signertypes.Signer
+}
+
+// NewEthermanSigners creates a new instance of EthermanSigners
+func NewEthermanSigners(ctx context.Context, chainID uint64,
+	config []signertypes.SignerConfig) (*EthermanSigners, error) {
+	res := EthermanSigners{
+		chainID: chainID,
+		signers: make(map[common.Address]signertypes.Signer),
+	}
+	for i, signerConfig := range config {
+		signer, err := signer.NewSigner(ctx, chainID, signerConfig, fmt.Sprintf("signer-%d", i), nil)
+		if err != nil {
+			return nil, err
+		}
+		res.signers[signer.PublicAddress()] = signer
```

Comment:
Should we maybe have a check if someone defined multiple signers for the same address?